src/dump.c

```diff
@@ -2810,7 +2810,7 @@ static void jl_recache_other(void)
             _new = (jl_value_t*)jl_recache_method_instance((jl_method_instance_t*)o, i);
         }
         else {
-            assert(0);
+            abort();
```

Comment:
should this have any message indicating what went wrong? silently aborting isn't very nice